src/preamble.js

```diff
@@ -2401,7 +2403,13 @@ function integrateWasmJS(Module) {
   Module['asmPreload'] = Module['asm'];
 
   // Memory growth integration code
+
+  var defaultReallocBuffer = Module['reallocBuffer'];
+
   Module['reallocBuffer'] = function(size) {
+    if (finalMethod === 'asmjs') {
```

Comment:
`finalMethod` is a somewhat generic looking name, and it's also in global scope. Perhaps it would be better to call it `binaryenMethod` or something similar?